src/crypto.ts

```diff
@@ -1,11 +1,69 @@
-import * as blake2 from "blake2";
-
-// hashAll takes a list of string parameters and hashes them using the 'blake2b'
-// hasing function. The return value is a hex representation of that hash.
-export function hashAll(...args: string[]): string {
-  const h = blake2.createHash("blake2b");
-  for (const arg of args) {
-    h.update(Buffer.from(arg));
+import blake from "blakejs";
+import { RegistryValue } from "./registry";
+import { FileID } from "./skydb";
+
+// NewHash returns a blake2b 256bit hasher.
+function NewHash() {
+  return blake.blake2bInit(32, null);
+}
+
+// HashAll takes all given arguments and hashes them.
+export function HashAll(...args: any[]): Uint8Array {
+  const h = NewHash();
+  for (let i = 0; i < args.length; i++) {
+    blake.blake2bUpdate(h, args[i]);
+  }
+  return blake.blake2bFinal(h);
+}
+
+// HashRegistryValue hashes the given registry value and returns it as a hex
+// encoded string
+export function HashRegistryValue(registryValue: RegistryValue): string {
+  return Buffer.from(
+    HashAll(
+      hexToUint8Array(registryValue.tweak),
+      encodeString(registryValue.data),
+      encodeNumber(registryValue.revision)
+    )
+  ).toString("hex");
+}
+
+// HashFileID hashes the given fileID and returns it as a hex encoded string
+export function HashFileID(fileID: FileID): string {
+  return Buffer.from(
+    HashAll(
+      encodeNumber(fileID.version),
+      encodeString(fileID.applicationID),
+      encodeNumber(fileID.fileType),
+      encodeString(fileID.filename)
+    )
+  ).toString("hex");
+}
+
+// asciiToUint8Array converts a string to a uint8 array
+function asciiToUint8Array(str: string): Uint8Array {
```

Comment:
I renamed but have not moved it. Because I changed the implementation, I only wanted my new code to use that function, instead of potentially breaking your encryption code. It is not exported so we can (re)move this at any time. I would rather keep it here until the SDK is out if everything works the way it is now.